apps/server/src/repositories/learnroom/course.repo.integration.spec.ts

```diff
@@ -78,65 +89,132 @@ describe('course repo', () => {
-			const result = await repo.findAllByUserId(helper.getFirstUser().id);
+			await em.persistAndFlush([course1, course2]);
+			em.clear();
 
-			const expectedResult = [courses, 2];
-			expect(result).toEqual(expectedResult);
+			const [result, count] = await repo.findAllByUserId(teacher.id);
+
+			expect(checkEqualIds(result, [course1, course2])).toEqual(true);
+			expect(count).toEqual(2);
 		});
 
 		it('should return course of students', async () => {
-			const helper = new LearnroomTestHelper();
-			const courses = [helper.createStudentCourse(), helper.createStudentCourse()];
+			const student = new UserTaskInfo({ firstName: 'John', lastName: 'Doe' });
+			await em.persistAndFlush(student);
+			const course1 = new Course({ name: 'course #1', schoolId: new ObjectId(), studentIds: [student._id] });
+			const course2 = new Course({ name: 'course #2', schoolId: new ObjectId(), studentIds: [student._id] });
 
-			await em.persistAndFlush(courses);
-			const result = await repo.findAllByUserId(helper.getFirstUser().id);
+			await em.persistAndFlush([course1, course2]);
+			em.clear();
 
-			const expectedResult = [courses, 2];
-			expect(result).toEqual(expectedResult);
+			const [result, count] = await repo.findAllByUserId(student.id);
+
+			expect(checkEqualIds(result, [course1, course2])).toEqual(true);
+			expect(count).toEqual(2);
 		});
 
 		it('should return course of substitution teachers', async () => {
-			const helper = new LearnroomTestHelper();
-			const courses = [helper.createTeacherCourse(), helper.createTeacherCourse()];
-
-			await em.persistAndFlush(courses);
-			const result = await repo.findAllByUserId(helper.getFirstUser().id);
-
-			const expectedResult = [courses, 2];
-			expect(result).toEqual(expectedResult);
+			const subTeacher = new UserTaskInfo({ firstName: 'John', lastName: 'Doe' });
+			await em.persistAndFlush(subTeacher);
+			const course1 = new Course({
+				name: 'course #1',
+				schoolId: new ObjectId(),
+				substitutionTeacherIds: [subTeacher._id],
+			});
+			const course2 = new Course({
+				name: 'course #2',
+				schoolId: new ObjectId(),
+				substitutionTeacherIds: [subTeacher._id],
+			});
+
+			await em.persistAndFlush([course1, course2]);
+			em.clear();
+
+			const [result, count] = await repo.findAllByUserId(subTeacher.id);
+
+			expect(checkEqualIds(result, [course1, course2])).toEqual(true);
+			expect(count).toEqual(2);
 		});
 
 		it('should handle mixed roles in courses', async () => {
-			const helper = new LearnroomTestHelper();
-			const courses = [helper.createStudentCourse(), helper.createTeacherCourse(), helper.createSubstitutionCourse()];
-
-			await em.persistAndFlush(courses);
-			const result = await repo.findAllByUserId(helper.getFirstUser().id);
-
-			const expectedResult = [courses, 3];
-			expect(result).toEqual(expectedResult);
+			const user = new UserTaskInfo({ firstName: 'John', lastName: 'Doe' });
+			await em.persistAndFlush(user);
+			const course1 = new Course({
```

Comment:
Test should point only the important test case, in this case boilerplate code should hide all the time in tests.

Reply:
when only using this id's in references, half-page length for an it is fine for me. the helper on the otherside is very unspecific. can something be moved up into a beforeAll?

Reply:
that's what factories are for. we should use them and optimize the boilerplate code as soon as we are more confident how to structure these tests. this was just a first step to a more explicit test setup.

Reply:
For me it is a step back in the point how many place i must change if courses are changed. 
It is also a step back by replacing it, because we lost the one linked place.
And the important point is ..we have a ticket that include the switch from helpers to factories ..but this is not this ticket.
This only revert the first step on the way to the factories and bring us nothing.

Reply:
the tests had to be adapted because the entity relations had changed. i chose the easiest possible approach for that - just using the plain entity constructors. and that - as i said - can be automatted, e.g. by using factories. yes that will be done in another ticket. as far as i can see we have currently not much more code for test setup than before. and that without any additional measures. plus, the tests are easier to understand imho. 
when en entity has to be change it's obvious that all places where the entity is used are subject to change not only the tests. but that's natural (refactoring).